packages/sdk-auth/src/auth.js

```diff
@@ -313,12 +329,12 @@ export default class SdkAuth {
       headers,
     })
 
-    const request = SdkAuth._buildRequest(_config, uri)
+    let request = SdkAuth._buildRequest(_config, uri)
     request.body = body || '' // let user to build their own body
 
     if (credentials)
-      request.body = SdkAuth._appendUserCredentialsToBody(
-        request.body,
+      request = SdkAuth._appendUserCredentialsToBody(
+        request,
```

Comment:
I see, we keep this pattern throughout 👍

Reply:
The conditionals are a bit hard to improve without better patterns of compositions. I will adjust those which don't involve them.